components/ILIAS/Certificate/classes/User/class.ilUserCertificate.php

```diff
@@ -40,6 +42,7 @@ public function __construct(
         private int $version,
         private readonly string $iliasVersion,
         private readonly bool $currentlyActive,
+        private readonly CertificateId $certificate_id,
```

Comment:
Maybe we should re-arrange the properties. IMO `CertificateId` should be the 1st parameter, `id` should be the 2nd parameter (and be renamed, only in this class, to `persistence_id` or similar).

Reply:
id can't be the second parameter because it's nullable

Reply:
should I still make CertificateId the first parameter?